src/topic_based_system.cpp

```diff
@@ -133,20 +133,39 @@ CallbackReturn TopicBasedSystem::on_init(const hardware_interface::HardwareInfo&
     }
     return default_value;
   };
+  // Load parameters from the hardware interface
+  time_from_start_ = std::stod(get_hardware_parameter("time_from_start", "0.1"));
+  position_cmd_only_ = get_hardware_parameter("position_cmd_only", "false") == "true";
+  std::string joint_commands_type_str = get_hardware_parameter("joint_commands_type", "sensor_msgs/JointState");
+  try
+  {
+    joint_commands_type_ = stringToCommandType(joint_commands_type_str);
+  }
+  catch (const std::exception& e)
+  {
+    RCLCPP_ERROR(node_->get_logger(), "%s", e.what());
+    return CallbackReturn::ERROR;
+  }
+
+  block_joint_command_threshold_ = std::stod(get_hardware_parameter("block_joint_command_threshold", "0.5"));
+  trigger_joint_command_threshold_ = std::stod(get_hardware_parameter("trigger_joint_command_threshold", "1e-5"));
 
   // Add random ID to prevent warnings about multiple publishers within the same node
   rclcpp::NodeOptions options;
   options.arguments({ "--ros-args", "-r", "__node:=topic_based_ros2_control_" + info_.name });
 
   node_ = rclcpp::Node::make_shared("_", options);
 
-  if (auto it = info_.hardware_parameters.find("trigger_joint_command_threshold"); it != info_.hardware_parameters.end())
+  if (joint_commands_type_ == TopicBasedCmdType::JOINT_STATE)
```

Comment:
Still use if?

Reply:
Oh i think it is fine to use `if `here. But if  in `write()`, it is better to use `switch`